http-server-netty/src/main/java/io/micronaut/http/server/netty/configuration/NettyHttpServerConfiguration.java

```diff
@@ -545,6 +545,22 @@ public void setPcapLoggingPathPattern(String pcapLoggingPathPattern) {
         this.pcapLoggingPathPattern = pcapLoggingPathPattern;
     }
 
+    /**
+     * Get the explicit netty listener configurations, or {@code null} if they should be implicit.
+     * @return The listeners
+     */
+    public Map<String, NettyListenerConfiguration> getListeners() {
+        return listeners;
+    }
+
+    /**
+     * Set the explicit netty listener configurations, or {@code null} if they should be implicit.
+     * @param listeners The listeners
+     */
+    public void setListeners(Map<String, NettyListenerConfiguration> listeners) {
+        this.listeners = listeners;
+    }
```

Comment:
As it is written now, Jackson will be used to convert from a `Map` to an `Object` (see attached debug session screenshot):

<img width="992" alt="Screenshot 2022-05-06 at 14 56 19" src="https://user-images.githubusercontent.com/66626/167135838-79d45ab9-86a7-4fef-aa5d-adfae4cdffe7.png">

This would require reflection, unless the type is annotated with `@Introspected` then perhaps not. If you write it as a inner class you benefit from several advantages:

1. No reflection config or `@Introspected` is needed
2. The IDE will present code completion to the user when they write listener configurations since we will generate the necessary configuration metadata

Reply:
@graemerocher like this?

Reply:
Hmm the change added a spurious test failure, seems like EachProperty does not preserve order?

Reply:
hmm the functionality is dependent on order of YAML?

Reply:
mostly for old api, getPort returns the first listener

Reply:
I am not sure this is good. Some config forms (like java properties) we support don't support ordering at all and we have never required ordering in any config we have had before.

Can you rewrite the test to not depend on order?

Reply:
ok will do. it means theres no way to influence what getPort will return if you use the new listener config, but i guess it's not a big deal